File: sdks/ruby/examples/send_sms.rb
# frozen_string_literal: true

# Send an SMS with the Pushfy Ruby SDK.
#   ruby examples/send_sms.rb
# Requires PUSHFY_API_TOKEN in your environment.

require_relative "../lib/pushfy"

def main
  pushfy = Pushfy::Client.new(api_token: ENV["PUSHFY_API_TOKEN"])

  result = pushfy.sms.send(
    to: "5511999999999",
    text: "Hello from the Pushfy Ruby SDK",
    ext_id: "demo-#{Time.now.to_i}"
  )
  puts "Accepted: #{result.inspect}"
rescue Pushfy::RateLimitError
  warn "Rate limited — back off and retry."
  exit 1
rescue Pushfy::PushfyError => e
  warn "Failed: #{e.status} #{e.code} #{e.response.inspect}"
  exit 1
end

main

File: sdks/node/examples/send-sms.js
'use strict';

// Send an SMS with the Pushfy Node SDK.
//   node examples/send-sms.js
// Requires PUSHFY_API_TOKEN in your environment.

const { Pushfy, RateLimitError } = require('../src');

async function main() {
  const pushfy = new Pushfy({ apiToken: process.env.PUSHFY_API_TOKEN });

  try {
    const result = await pushfy.sms.send({
      to: '5511999999999',
      text: 'Hello from the Pushfy Node SDK 👋',
      extId: 'demo-' + Date.now(),
    });
    console.log('Accepted:', result);
  } catch (err) {
    if (err instanceof RateLimitError) console.error('Rate limited — back off and retry.');
    else console.error('Failed:', err.status, err.code, err.response);
    process.exit(1);
  }
}

main();
